auth/src/main/java/org/aerogear/mobile/security/SecurityCheckMetricPublisher.java

```diff
@@ -0,0 +1,36 @@
+package org.aerogear.mobile.security;
+
+
+import org.aerogear.mobile.core.metrics.MetricsService;
+import org.aerogear.mobile.security.metrics.SecurityCheckResultMetric;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This object will manage the communication with the metric service, batching the results to be published.
+ */
+class SecurityCheckMetricPublisher implements SecurityCheckExecutorListener {
+
+    private final MetricsService metricsService;
+    private final List<SecurityCheckResultMetric> metrics = new ArrayList<>();
+
+    /**
+     * Builds the object.
+     *
+     * @param metricsService metric service to be used to publish the metrics
+     */
+    SecurityCheckMetricPublisher(final MetricsService metricsService) {
```

Comment:
We could pass an executor here, to publish the metrics asynchronously. WDYT?

Reply:
I think the metrics service itself should choose the right executor. For now, let's not worry about it.

Reply:
:+1:  I think this would be done at the same time as the metrics service being improved to batch sequential requests by itself.